src/time-picker/TimePicker.js

```diff
@@ -136,13 +139,17 @@ export const filterTime = ({ originalInput, minTime, maxTime }) => time => {
   return lteTime(minTime, time) && lteTime(time, maxTime) && startsWith(time, originalInput);
 };
 
-export const makeOptions = ({ minTime, maxTime, timeFormat }, inputValue) => {
+export const makeOptions = ({ minTime, maxTime, timeFormat, userValue }, inputValue) => {
   const time = parseInTimeFormat(inputValue, timeFormat);
-  const timeList = time === inputError ? [] : createTimeList(time, timeFormat);
+  //make sure the userValue has timeFormat so in H12 the day period will be shown
+  if ( userValue && userValue !== inputError ) {
+    userValue.timeFormat = timeFormat;
+  }
```

Comment:
if you don't set the TimeFormat, it will consider it as H24 and will show the option as:

![image](https://cloud.githubusercontent.com/assets/22367312/24291539/aafb42ee-1089-11e7-922a-9db16902f4f2.png)


it is used in 

`const formatTime = ({ hours, minutes, timeFormat }) => (
  timeFormat === H12 ? formatTime12({ hours, minutes }) : formatTime24({ hours, minutes })
);`